### PR TITLE
Add destructive job for vSphere 4.14

### PR DESCRIPTION
**Pull request by sgaoshang**

### CODE REVIEW

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 1]
```diff
@@ -3400,8 +3425,20 @@ tests:
...
+    cluster_profile: vsphere
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-ipi and @network-ovnkubernetes and @destructive'
+      FIPS_ENABLED: "true"
+      TAG_VERSION: '@4.14'
+      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
+    test:
+    - chain: openshift-e2e-test-qe-destructive
+    workflow: cucushift-installer-rehearse-vsphere-ipi-disk-encryption-tang
+- as: vsphere-ipi-external-lb-pre-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3440]: Based on https://github.com/openshift/release/pull/42857 is deprecating this job, I think you can untouch with this job.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 2]
```diff
@@ -3410,8 +3447,20 @@ tests:
...
-  cron: 43 10 8,23 * *
+- as: vsphere-ipi-external-lb-pre-f28-destructive
+  cron: 21 5 13 * *
+  steps:
+    cluster_profile: vsphere
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-ipi and @network-ovnkubernetes and not @fips
+        and @destructive'
+      TAG_VERSION: '@4.14'
+      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3458]: Wrong `TEST_FILTERS`?

Comment (sgaoshang) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3458]: Maybe I make a mistake, is this `TEST_FILTERS` only apply to disconnected job?

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3458]: I think the `~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&` is only used for disconnected install.

Comment (sgaoshang) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3458]: If so, I'll remove it from other connected job too.

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3458]: For a common connected install, I think we do not need to set it.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 3]
```diff
@@ -3420,8 +3469,19 @@ tests:
...
-- as: vsphere-ipi-ovn-ipsec-f14
-  cron: 24 23 14,29 * *
+- as: vsphere-ipi-ovn-dualstack-f28-destructive
+  cron: 42 4 29 * *
+  steps:
+    cluster_profile: vsphere
+    env:
+      E2E_RUN_TAGS: '@vsphere-ipi and @network-ovnkubernetes and not @fips and @destructive'
+      TAG_VERSION: '@4.14'
+      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3479]: Wrong TEST_FILTERS?

I think the ~ChkUpgrade&;~NonPreRelease&;~Serial&;~Disruptive&;~ConnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly& is only used for disconnected install.

For a common connected install, I think we do not need to set it.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 4]
```diff
@@ -3593,8 +3800,20 @@ tests:
...
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-vsphere-upi-disconnecting-network
-- as: vsphere-upi-encrypt-f14
-  cron: 19 17 2,17 * *
+- as: vsphere-upi-disconnecting-network-f28-destructive
+  cron: 57 8 28 * *
+  steps:
+    cluster_profile: vsphere-connected
+    env:
+      E2E_RUN_TAGS: '@amd64 and @vsphere-upi and @disconnected and @network-openshiftsdn
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3808]: by a chance, I saw `@network-openshiftsdn` is set here, do you mind to update it to `@network-ovnkubernetes` together with `vsphere-upi-disconnecting-network-f28` job.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml [hunk 5]
```diff
@@ -3557,20 +3714,32 @@ tests:
...
     workflow: cucushift-installer-rehearse-vsphere-ipi-zones
-- as: vsphere-upi-disconnected-sdn-workers-rhcos-rhel8-f14
-  cron: 8 20 10,25 * *
+- as: vsphere-upi-disconnected-sdn-rhcos-rhel8-f28
+  cron: 33 22 30 * *
   steps:
     cluster_profile: vsphere-dis
     env:
-      E2E_RUN_TAGS: '@amd64 and @vsphere-upi and @disconnected and @network-openshiftsdn
+      E2E_RUN_TAGS: '@amd64 and @vsphere-upi and @disconnected and @network-ovnkubernetes
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-nightly.yaml:3722]: hmm, one this line, `@network-openshiftsdn` is right.

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Pull request merged**